coralogix/dashboard_widgets/hexagon.go

```diff
@@ -0,0 +1,813 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dashboardwidgets
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"terraform-provider-coralogix/coralogix/utils"
+
+	cxsdk "github.com/coralogix/coralogix-management-sdk/go"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/objectvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+var (
+	DashboardSchemaToProtoHexagonAggregation = map[string]cxsdk.HexagonMetricAggregation{
+		"unspecified": cxsdk.HexagonMetricAggregationUnspecified,
+		"last":        cxsdk.HexagonMetricAggregationLast,
+		"min":         cxsdk.HexagonMetricAggregationMin,
+		"max":         cxsdk.HexagonMetricAggregationMax,
+		"avg":         cxsdk.HexagonMetricAggregationAvg,
+		"sum":         cxsdk.HexagonMetricAggregationSum,
+	}
+
+	DashboardProtoToSchemaHexagonMetricAggregation = utils.ReverseMap(DashboardSchemaToProtoHexagonAggregation)
+	DashboardValidHexagonMetricAggregations        = utils.GetKeys(DashboardSchemaToProtoHexagonAggregation)
+)
+
+type HexagonModel struct {
+	CustomUnit    types.String       `tfsdk:"custom_unit"`
+	LegendBy      types.String       `tfsdk:"legend_by"`
+	Decimal       types.Number       `tfsdk:"decimal"`
+	DataModeType  types.String       `tfsdk:"data_mode_type"`
+	Thresholds    types.Set          `tfsdk:"thresholds"` //HexagonThresholdModel
+	ThresholdType types.String       `tfsdk:"threshold_type"`
+	Min           types.Number       `tfsdk:"min"`
+	Max           types.Number       `tfsdk:"max"`
+	Unit          types.String       `tfsdk:"unit"`
+	Legend        *LegendModel       `tfsdk:"legend"`
+	Query         *HexagonQueryModel `tfsdk:"query"`
+	TimeFrame     *TimeFrameModel    `tfsdk:"time_frame"`
+}
+
+type HexagonQueryModel struct {
+	Logs      *HexagonQueryLogsModel `tfsdk:"logs"`
+	Metrics   *QueryMetricsModel     `tfsdk:"metrics"`
+	Spans     *QuerySpansModel       `tfsdk:"spans"`
+	DataPrime *DataPrimeModel        `tfsdk:"data_prime"`
+}
+
+type HexagonQueryLogsModel struct {
+	LuceneQuery types.String          `tfsdk:"lucene_query"`
+	GroupBy     types.List            `tfsdk:"group_by"`    //ObservationFieldModel
+	Aggregation *LogsAggregationModel `tfsdk:"aggregation"` //AggregationModel
```

Comment:
The comment suppose to refer to the underlying type when using types implicitly (types.Object, types.List etc...). So it's irrelevant and misleading here.